src/Workload/Microsoft.Maui.Sdk/Sdk/Microsoft.Maui.Core.Sdk.After.targets

```diff
@@ -5,6 +5,12 @@
     <MSBuildWarningsAsMessages Condition=" '$(TargetPlatformIdentifier)' == 'Android' ">$(MSBuildWarningsAsMessages);XA4218</MSBuildWarningsAsMessages>
   </PropertyGroup>
 
-  <Import Project="WinUI.targets" Condition=" '$(TargetPlatformIdentifier)' == 'windows' " />
+  <!--
+    This double check seems excessive, but importing the WinUI.targets when building for windows is not
+    sufficient. The WASDK targets assume everything is WinUI and thus just passes along the TFM of the app.
+    As a result, if you have a net6.0 class library, the app will call MSBuild on that library - with the Windows TFM!
+    This results in the $(TargetPlatformIdentifier) condition being met - even though there are no WASK targets to run!
+  -->
+  <Import Project="WinUI.targets" Condition=" '$(TargetPlatformIdentifier)' == 'windows' and '$(WindowsAppSDKWinUI)' == 'true'" />
```

Comment:
Make sure that when this target is imported it is BOTH a windows project AND it is referencing the WASDK nuget. If MSBuild is run with a TargetFramework of windows, the first condition will become true, but the second one will not because the real project will not have referenced the WASDK nuget.

And this is fine because all the targets in the WinUI.targets are workarounds for windows AND WASDK nuget together.